plugins/reindex/src/main/java/org/elasticsearch/plugin/reindex/BulkIndexByScrollResponse.java

```diff
@@ -0,0 +1,163 @@
+import static org.elasticsearch.plugin.reindex.BulkIndexByScrollResponse.Fields.VERSION_CONFLICTS;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+import org.elasticsearch.action.ActionResponse;
+import org.elasticsearch.action.bulk.BulkItemResponse.Failure;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.common.xcontent.ToXContent;
+import org.elasticsearch.common.xcontent.XContentBuilder;
+import org.elasticsearch.common.xcontent.XContentBuilderString;
+
+/**
+ * Response used for actions that index many documents using a scroll request.
+ */
+public class BulkIndexByScrollResponse extends ActionResponse implements ToXContent {
+    private long took;
+    private long updated;
+    private int batches;
+    private long versionConflicts;
+    private long noops;
+    private List<Failure> failures;
+
+    public BulkIndexByScrollResponse() {
+    }
+
+    public BulkIndexByScrollResponse(long took, long updated, int batches, long versionConflicts, long noops, List<Failure> failures) {
+        this.took = took;
+        this.updated = updated;
+        this.batches = batches;
+        this.versionConflicts = versionConflicts;
+        this.noops = noops;
+        this.failures = unmodifiableList(failures);
+    }
+
+    public long getTook() {
+        return took;
+    }
+
+    public long getUpdated() {
+        return updated;
+    }
+
+    public int getBatches() {
+        return batches;
+    }
+
+    public long getVersionConflicts() {
+        return versionConflicts;
+    }
+
+    public long getNoops() {
+        return noops;
+    }
+
+    /**
+     * All recorded failures.
+     */
+    public List<Failure> getFailures() {
+        return failures;
+    }
+
+    @Override
+    public void writeTo(StreamOutput out) throws IOException {
+        super.writeTo(out);
+        out.writeVLong(took);
+        out.writeVLong(updated);
+        out.writeVInt(batches);
+        out.writeVLong(versionConflicts);
+        out.writeVLong(noops);
+        out.writeVInt(failures.size());
+        for (Failure failure: failures) {
+            failure.writeTo(out);
+        }
+    }
+
+    @Override
+    public void readFrom(StreamInput in) throws IOException {
+        super.readFrom(in);
+        took = in.readVLong();
+        updated = in.readVLong();
+        batches = in.readVInt();
+        versionConflicts = in.readVLong();
+        noops = in.readVLong();
+        int failureCount = in.readVInt();
+        List<Failure> failures = new ArrayList<>(failureCount);
+        for (int i = 0; i < failureCount; i++) {
+            Failure failure = new Failure();
+            failure.readFrom(in);
+            failures.add(failure);
+        }
+        this.failures = unmodifiableList(failures);
+    }
+
+    static final class Fields {
```

Comment:
can we just make these `XContentBuilderString` static a `static final String` in the enclosing class? I know we do this everywhere but it's really just confusing and not helpful.